src/Model/Reports.php

```diff
@@ -0,0 +1,869 @@
+		'admintools.pluginEnable'    => 'PANOPTICON_REPORTS_LBL_SITEACTION_ADMINTOOLS_PLUGIN_ENABLE',
+		'admintools.unblockMyIP'     => 'PANOPTICON_REPORTS_LBL_SITEACTION_ADMINTOOLS_UNBLOCK_MY_IP',
+		'akeebabackup.delete'        => 'PANOPTICON_REPORTS_LBL_SITEACTION_AKEEBABACKUP_DELETE',
+		'akeebabackup.deleteFiles'   => 'PANOPTICON_REPORTS_LBL_SITEACTION_AKEEBABACKUP_DELETEFILES',
+	];
+
+	/** @inheritdoc */
+	public function __construct(Container $container = null)
+	{
+		$this->tableName   = '#__reports';
+		$this->idFieldName = 'id';
+
+		parent::__construct($container);
+
+		$this->addBehaviour('filters');
+	}
+
+	/**
+	 * Returns a new object for a Core Update Found event
+	 *
+	 * @param   int          $site_id     The side ID
+	 * @param   string|null  $oldVersion  The old version of the CMS
+	 * @param   string|null  $newVersion  The latest available version of the CMS
+	 *
+	 * @since   1.0.4
+	 */
+	public static function fromCoreUpdateFound(
+		int $site_id, ?string $oldVersion, ?string $newVersion
+	): static
+	{
+		/** @var static $item */
+		$item             = Factory::getContainer()->mvcFactory->makeTempModel('reports');
+		$item->site_id    = $site_id;
+		$item->created_on = 'now';
+		$item->created_by = 0;
+		$item->action     = ReportAction::CORE_UPDATE_FOUND;
+		$item->context    = [
+			'oldVersion' => $oldVersion,
+			'newVersion' => $newVersion,
+		];
+
+		return $item;
+	}
+
+	/**
+	 * Returns a new object for a Core Update Installed event
+	 *
+	 * @param   int          $site_id         The side ID
+	 * @param   string|null  $oldVersion      The old version of the CMS
+	 * @param   string|null  $newVersion      The latest available version of the CMS
+	 * @param   bool|null    $success         Has the update completed successfully?
+	 * @param   mixed        $furtherContext  Additional context information
+	 *
+	 * @since   1.0.4
+	 */
+	public static function fromCoreUpdateInstalled(
+		int $site_id, ?string $oldVersion, ?string $newVersion, ?bool $success = null, mixed $furtherContext = null
+	): static
+	{
+		/** @var static $item */
+		$item             = Factory::getContainer()->mvcFactory->makeTempModel('reports');
+		$item->site_id    = $site_id;
+		$item->created_on = 'now';
+		$item->created_by = 0;
+		$item->action     = ReportAction::CORE_UPDATE_INSTALLED;
+		$item->context    = [
+			'oldVersion' => $oldVersion,
+			'newVersion' => $newVersion,
+			'success'    => $success,
+			'context'    => self::furtherContextAsArrayOrNull($furtherContext),
+		];
+
+		return $item;
+	}
+
+	/**
+	 * Returns a new object for an Extension Update Found event
+	 *
+	 * @param   int          $site_id        The side ID
+	 * @param   string       $extensionKey   The extension key, e.g. com_example
+	 * @param   string       $extensionName  The human-readable extension name, e.g. "An Example"
+	 * @param   string|null  $oldVersion     The old version of the CMS
```

Comment:
shouldn't this (and similar) be "the old versi9on of the extension"

Reply:
Yup. Wanna make a PR for it?

Reply:
ok - will do when I get back from the hsp[ital

Reply:
Thank you — and I hope it all goes well for you and your family.